roles/setup_server/tasks/main.yaml

```diff
@@ -31,6 +31,22 @@
       ansible.builtin.include_role:
         name: fedora.linux_system_roles.rhc
       when:
+        - not setup_server_beta_repos
+        - 'ansible_distribution == "RedHat" and not setup_server_rhsatellite'
+        - '"rh-amazon-rhui-client" not in ansible_facts["packages"]'
+        - '"rhui-azure-rhel9" not in ansible_facts["packages"]'
+        - '"rhui-azure-rhel8" not in ansible_facts["packages"]'
+
+    - name: Setup Beta Repositories
+      vars:
+        rhc_repositories:
+          - {name: "*", state: disabled}
+          - {name: "rhel-{{ ansible_distribution_major_version }}-for-x86_64-appstream-beta-rpms", state: enabled}
+          - {name: "rhel-{{ ansible_distribution_major_version }}-for-x86_64-baseos-beta-rpms", state: enabled}
```

Comment:
These should be conditionalized on the `$arch` of the running system.  See: #283

Reply:
Good catch thanks!